projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -194,11 +195,39 @@ public function capture_product_view() {
 		);
 	}
 
+	/**
+	 * Get the selected shipping option for a cart item. If the name cannot be found in the options table, the method's
+	 * ID will be used.
+	 *
+	 * @param string $cart_item_key the cart item key.
+	 *
+	 * @return mixed|void
+	 */
+	public function get_shipping_option_for_item( $cart_item_key ) {
+		$packages         = wc()->shipping()->get_packages();
+		$selected_options = wc()->session->get( 'chosen_shipping_methods' );
+		foreach ( $packages as $package_id => $package ) {
+			foreach ( $package['contents'] as $package_item ) {
+				if ( $package_item['key'] === $cart_item_key ) {
+					$selected_rate_id = $selected_options[ $package_id ];
+					$method_key_id    = str_replace( ':', '_', $selected_rate_id );
+					$option_name      = 'woocommerce_' . $method_key_id . '_settings';
```

Comment:
Should we sanitize `$method_key_id`, to avoid any unwanted values that may be passed maliciously?

Reply:
I don't know if this would be possible because this would be defined by the shipping method extension, however I added `sanitize_text_field` just in case. Is that what you had in mind?

2dd51b597fa61e84832b0f74b6b379463885b71e